### PR TITLE
Fix _all_docs call where keys is an empty list

### PR DESCRIPTION
**Pull request by smithsz**
## What
Fix `/_all_docs` call where `keys=[]`.

## How
The `get_docs` common function was incorrectly dropping the `keys` parameter if the value of `keys` was an empty list.

## Testing
Includes additional unit tests.

## Issues
- #289

### CODE REVIEW

File: tests/unit/common_util_tests.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+        resp = util.get_docs(m_sess, url, keys=keys)
+
+        m_post.assert_called_once_with(
+            url,
+            data='{"keys": []}',
+            headers={'Content-Type': 'application/json'},
+            params={}
+        )
+
+        self.assertEquals(200, resp.status_code)
```
Comment (alfinkel) [inline tests/unit/common_util_tests.py]: Why would you choose to write mocked tests when we've been writing tests using couchdb tests all along?  Just thinking that it makes the library appear cleaner if our tests are structured and written consistently.  I feel like if you're going to start a CommonUtilsTests module you'll need to provide tests for all of the functions in the `_common_utils` module which obviously would be out of scope for this PR.  Perhaps remove these tests and just add test(s) targetting couchdb as part of the Database tests that ensures this `get_doc` method is working?  I open to other suggestions, I just think that right now these tests are a bit misaligned with the rest of our test suite in this library.

Comment (smithsz) [inline tests/unit/common_util_tests.py]: righto, I'll add another all_docs test in database_tests using an empty key list and remove these.

**Comment by alfinkel** [comment]
Looks like the CI should right itself once #292 is merged into master and this PR is rebased based on that.  Obviously my +1 is contingent on the CI passing.
